rust/azure_iot_operations_services/src/azure_device_registry/client.rs

```diff
@@ -1,2 +1,1652 @@
+
+        let response = self
+            .get_device_command_invoker
+            .invoke(get_device_request)
+            .await
+            .map_err(ErrorKind::from)?;
+        Ok(response.payload.device.into())
+    }
+
+    /// Updates a Device's status in the Azure Device Registry service.
+    ///
+    /// # Arguments
+    /// * `device_name` - The name of the Device.
+    /// * `inbound_endpoint_name` - The name of the inbound endpoint.
+    /// * [`DeviceStatus`] - All status information for the device.
+    /// * `timeout` - The duration until the Client stops waiting for a response to the request, it is rounded up to the nearest second.
+    ///
+    /// Returns the updated [`Device`] once updated.
+    ///
+    /// # Errors
+    /// [`struct@Error`] of kind [`InvalidRequestArgument`](ErrorKind::InvalidRequestArgument)
+    /// if timeout is 0.
+    ///
+    /// [`struct@Error`] of kind [`AIOProtocolError`](ErrorKind::AIOProtocolError)
+    /// if device, inbound endpoint are invalid.
+    ///
+    /// [`struct@Error`] of kind [`AIOProtocolError`](ErrorKind::AIOProtocolError)
+    /// if there are any underlying errors from the AIO RPC protocol.
+    pub async fn update_device_plus_endpoint_status(
+        &self,
+        device_name: String,
+        inbound_endpoint_name: String,
+        status: DeviceStatus,
+        timeout: Duration,
+    ) -> Result<Device, Error> {
+        let status_payload = adr_name_gen::UpdateDeviceStatusRequestPayload {
+            device_status_update: status.into(),
+        };
+        let update_device_status_request =
+            adr_name_gen::UpdateDeviceStatusRequestBuilder::default()
+                .payload(status_payload)
+                .map_err(ErrorKind::from)?
+                .topic_tokens(Self::get_topic_tokens(device_name, inbound_endpoint_name))
+                .timeout(timeout)
+                .build()
+                .map_err(ErrorKind::from)?;
+        let response = self
+            .update_device_status_command_invoker
+            .invoke(update_device_status_request)
+            .await
+            .map_err(ErrorKind::from)?;
+        Ok(response.payload.updated_device.into())
+    }
+
+    /// Starts observation of any Device updates from the Azure Device Registry service.
+    ///
+    /// Note: On cleanup, unobserve should always be called so that the service knows to stop sending notifications.
+    ///
+    /// # Arguments
+    /// * `device_name` - The name of the Device.
+    /// * `inbound_endpoint_name` - The name of the inbound endpoint.
+    /// * `timeout` - The duration until the Client stops waiting for a response to the request, it is rounded up to the nearest second.
+    ///
+    /// Returns the [`DeviceUpdateObservation`] if the observation was started successfully or [`Error`].
+    ///
+    /// # Errors
+    /// [`struct@Error`] of kind [`DuplicateObserve`](ErrorKind::DuplicateObserve)
+    /// if the receiver is already registered.
```

Comment:
```suggestion
    /// if the device is already being observed.
```
receiver and register aren't relevant terms for the user of this API